plugin/src/main/java/at/helpch/chatchat/ChatChatPlugin.java

```diff
@@ -44,18 +59,32 @@ public final class ChatChatPlugin extends JavaPlugin {
     private @NotNull
     final Database database = new GsonDatabase(this);
     private @NotNull
-    final UsersHolder usersHolder = new UsersHolder(this);
+    final UsersHolderImpl usersHolder = new UsersHolderImpl(this);
```

Comment:
same again with these and their respective getters. i know that for some of these we do use some implementation details, but for those that don't need the implementation details we should abstract the return type.

Reply:
As we were talking in discord, I think this is a better way of doing it. The API should only give out interfaces but the implementation should use the implemented classes. This also helps in the future when we might want to add some extra stuff in our implementation that we don't want to expose in the API but we want to use in the plugin in which case we won't have to change back to this.